CC-35524 : Fix and Enable Integration tests for MongoDB Atlas connector

sonagaras opened the pull request:
## Fixed the integration tests
- There are total 85 integration tests in total. Fixed 7 of them.
- Current status is 81 Passing, 4 getting Skipped because they are for older versions only.

> - testBulkWriteOperationErrorWriteModelsIncludedInTheLog
>     - It was expecting content of key in the log, but we are not logging actual customer data from records.
>     - Fixed it by changing the expected message as per our logs.
> - testSinkCanHandleInvalidDocumentWhenErrorToleranceIsAll
>     - The reason is same, the test was expecting actual value in the error message.
> - testSourceEmitsNullValuesOnDelete
>     - We have changed the default value for DOCUMENT_KEY_AS_KEY_CONFIG as "false", while it is "true" in mongo repo.
>     - Fixed it by manually adding this config with value "true".
> - MongoSourceTaskIntegrationTest2
>     - 4 tests in this class, were explicitly looking for localhost.
>     - Changed them to have any connection string based on system property, or fall on default localhost.  

## Enabled docker instance for Mongodb to run the ITs
- MongoDB team written Integration tests was expecting a pre-setup mongodb instance to run the integration tests.
- Updated the code to use `testContainers`, which will automatically initiate docker instances of mongodb to run the test.

## Enabled integration tests in the semaphore pipeline
- The integration tests were disabled for MongoDB Atlas connector due to failures.
- Enabled them for semaphore, now on they will run upon every PR raise.

srishti-saraswat commented on src/integrationTest/java/com/mongodb/kafka/connect/mongodb/MongoDBHelper.java:
```diff
@@ -28,11 +28,22 @@
   private static final String DEFAULT_URI = "mongodb://localhost:27017";
   private static final String URI_SYSTEM_PROPERTY_NAME = "org.mongodb.test.uri";
   private static final String DEFAULT_DATABASE_NAME = "MongoKafkaTest";
+  private static final String MONGO_USERNAME = "test_username";
+  private static final String MONGO_PASSWORD = "test_password";
```
This should be in vault imo. Also, this is a public repo

aniketshrimal replied:
Since this is a docker instance and not an actual mongodb instance -> it should be okay to define these parameters in the code.

srishti-saraswat commented on src/integrationTest/java/com/mongodb/kafka/connect/mongodb/MongoDBHelper.java:
```diff
@@ -96,4 +110,15 @@ public ConnectionString getConnectionString() {
+
+  public static void startMongoContainer() {
+    MONGO_CONTAINER.start();
+    String mongoURI = MONGO_CONTAINER.getConnectionString()
```
can we have a null / running check for container?

sonagaras replied:
MONGO_CONTAINER is a constant, static final variable, that it cannot be null.
.start() method is idempotent, it will check and return in case of container already started.
Checked this calling .start() method twice immediately.

srishti-saraswat commented on src/integrationTest/java/com/mongodb/kafka/connect/mongodb/MongoDBHelper.java:
```diff
@@ -96,4 +110,15 @@ public ConnectionString getConnectionString() {
+  }
+
+  public static void closeMongoDbContainer() {
+    MONGO_CONTAINER.close();
```
maybe same check here if it would break

sonagaras replied:
Same for .close() it is also idempotent.

aniketshrimal commented:
@sonagaras The current active branch for mongodb is v1.16.x -> You can look to target this PR against that branch.

sonagaras committed:
Fixed some integration tests
Enabled docker instance for Mongodb to run the ITs
Enabled integration tests in the semaphore pipeline

aniketshrimal reviewed (approved):
LGTM @sonagaras !
